### ISSUE TITLE
Unable to define port in `addresses` block, nor enable HTTPS

### ISSUE THREAD

**Issue by mrcrilly** [hashicorp/consul#1727]
**TL;DR**
When using the following configuration, I am unable to get Consul to bind HTTP(S) to a particular port:

```JSON
{
  "addresses": {
    "http": "10.0.0.9:8080"
  }
}
```

The `https` key also does nothing, and does not enable TLS over HTTP as the configuration/documentation imply. The error thrown is:

```
server consul: ==> Starting Consul agent...
server consul: ==> Starting Consul agent RPC...
server consul: ==> Error starting http servers: Failed to get ClientListener address:port: Failed to parse IP: 10.0.0.9:8080
server systemd: consul.service: main process exited, code=exited, status=1/FAILURE
```

Here are the particulars of the server I am testing this one:

```Bash
# cat /etc/redhat-release
CentOS Linux release 7.2.1511 (Core) 
# consul version
Consul v0.6.3
Consul Protocol: 3 (Understands back to: 1)
```

**More Details**
I am unable to get Consul to honour my configuration in `consul.json` when I use the `addresses` nested object. Here is what the documentation states, at the time of writing, with regards to the `addresses` nested object:

```
addresses - This is a nested object that allows setting bind addresses. 

...

For TCP addresses, these should be in the form ip:port.

The following keys are valid:

    dns - The DNS server. Defaults to client_addr
    http - The HTTP API. Defaults to client_addr
    https - The HTTPS API. Defaults to client_addr
    rpc - The RPC endpoint. Defaults to client_addr
```

From this I understand that I can provide configuration as such:

```JSON
"addresses": {
    "https": "10.0.0.9:8080"
}
```

I also interrupt the document as saying: "An HTTP over TLS endpoint can be enabled by setting the `https` key in the `addresses` object."

As per [this page on encryption](https://www.consul.io/docs/agent/encryption.html), I see no mention of encrypting the HTTP API with TLS, only the Gossip and RPC protocols, which are different things.

Here is my configuration as it stands now:

```JSON
{
  "bootstrap_expect": 3,
  "bind_addr": "10.0.0.9",
  "addresses": {
    "https": "10.0.0.9"
  },
  "ca_file": "/etc/pki/tls/certs/ca-bundle.crt",
  "cert_file": "/etc/pki/tls/certs/our.crt",
  "key_file": "/etc/pki/tls/private/our.key",
  "datacenter": "domain-from-ansible",
  "data_dir": "/etc/consul",
  "log_level": "INFO",
  "server": true,
  "acl_datacenter": "domain-from-ansible",
  "acl_master_token": "random_token_here",
  "encrypt": "random_encryption_key_here"
}
```
This yields the following results from a `netstat -nplt`:

```Bash
# netstat -nplt
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 10.0.0.9:8300           0.0.0.0:*               LISTEN      16557/consul        
tcp        0      0 10.0.0.9:8301           0.0.0.0:*               LISTEN      16557/consul        
tcp        0      0 10.0.0.9:8302           0.0.0.0:*               LISTEN      16557/consul        
tcp        0      0 127.0.0.1:8400          0.0.0.0:*               LISTEN      16557/consul        
tcp        0      0 127.0.0.1:8500          0.0.0.0:*               LISTEN      16557/consul        
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1299/sshd           
tcp        0      0 127.0.0.1:8600          0.0.0.0:*               LISTEN      16557/consul        
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      1390/master         
tcp6       0      0 :::22                   :::*                    LISTEN      1299/sshd           
tcp6       0      0 ::1:25                  :::*                    LISTEN      1390/master 
```
As you can see, Consul has not implemented the configuration above, which clearly asked for `https` to be listening on `10.0.0.9`. If I change this to `http`, I get the following results:

```Bash
# netstat -nplt
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 10.0.0.9:8300           0.0.0.0:*               LISTEN      16959/consul        
tcp        0      0 10.0.0.9:8301           0.0.0.0:*               LISTEN      16959/consul        
tcp        0      0 10.0.0.9:8302           0.0.0.0:*               LISTEN      16959/consul        
tcp        0      0 127.0.0.1:8400          0.0.0.0:*               LISTEN      16959/consul        
tcp        0      0 10.0.0.9:8500           0.0.0.0:*               LISTEN      16959/consul        
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1299/sshd           
tcp        0      0 127.0.0.1:8600          0.0.0.0:*               LISTEN      16959/consul        
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      1390/master         
tcp6       0      0 :::22                   :::*                    LISTEN      1299/sshd           
tcp6       0      0 ::1:25                  :::*                    LISTEN      1390/master 
```

Now I have `http` listening on the correct IP. However, what about changing the port, like this: `http:8080`. What happens then?

```Bash
# netstat -nplt
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1299/sshd           
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      1390/master         
tcp6       0      0 :::22                   :::*                    LISTEN      1299/sshd           
tcp6       0      0 ::1:25                  :::*                    LISTEN      1390/master
```

And we have the following error:

```
server consul: ==> Starting Consul agent...
server consul: ==> Starting Consul agent RPC...
server consul: ==> Error starting http servers: Failed to get ClientListener address:port: Failed to parse IP: 10.0.0.9:8080
server systemd: consul.service: main process exited, code=exited, status=1/FAILURE
```

But as you'll remember from the direct quote of the documentation:

```
For TCP addresses, these should be in the form ip:port. 
```

**Comment by chuckyz** [comment]
Does this work with the following config...?

```
{
  "bootstrap_expect": 3,
  "bind_addr": "10.0.0.9",
  "addresses": {
    "https": "10.0.0.9"
  },
  "ports": {
     "https": "8080"
  },
  "ca_file": "/etc/pki/tls/certs/ca-bundle.crt",
  "cert_file": "/etc/pki/tls/certs/our.crt",
  "key_file": "/etc/pki/tls/private/our.key",
  "datacenter": "domain-from-ansible",
  "data_dir": "/etc/consul",
  "log_level": "INFO",
  "server": true,
  "acl_datacenter": "domain-from-ansible",
  "acl_master_token": "random_token_here",
  "encrypt": "random_encryption_key_here"
}
``` ?

**Comment by chuckyz** [comment]
The following works in a test VM:

```
{
  "addresses": {
	"http": "10.0.0.5"
  },
  "ports": {
	"http": 8080
  }
}
```

Started with...

`/usr/bin/consul agent -data-dir /usr/local/consul -server -bootstrap-expect=3 -config-dir /etc/consul.d`

Perhaps the docs should just be updated. :)

**Comment by mrcrilly** [comment]
I will confirm this shortly and do a PR for the documentation.

Thanks @chuckyz
